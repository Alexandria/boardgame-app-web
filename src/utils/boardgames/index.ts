export const boardgames = [
  {
    brdGameId: 1,
    name: "Catan",
    minPlayers: "3",
    maxPlayers: "4",
    avgPlayTime: "120",
    description:
      "In Catan (formerly The Settlers of Catan), players try to be the dominant force on the island of Catan by building settlements, cities, and roads. ",
    category: "Negotiation",
    minAge: "10",
    img:
      "https://cf.geekdo-images.com/thumb/img/g8LvJsd2oLAub6AEKN_Xpdoi8Sw=/fit-in/200x150/pic2419375.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/g8LvJsd2oLAub6AEKN_Xpdoi8Sw=/fit-in/200x150/pic2419375.jpg",
    bgGeekID: "13",
    createdAt: "2019-07-15T16:27:40.000Z",
    updatedAt: "2019-07-15T16:27:40.000Z",
  },
  {
    brdGameId: 2,
    name: "Basari",
    minPlayers: "3",
    maxPlayers: "4",
    avgPlayTime: "25",
    description:
      "Basari is a game of gem merchants competing in a marketplace&quot;: racing, collecting, trading, and predicting what the other merchants will do.",
    category: "Negotiation",
    minAge: "10",
    img:
      "https://cf.geekdo-images.com/original/img/xFRII0rLyA7eb2CDF_GT4BS-eLE=/0x0/pic48605.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/hBDymMYGWKQNbiuxDtqSQyYZDvQ=/fit-in/200x150/pic48605.jpg",
    bgGeekID: "14",
    createdAt: "2019-07-15T16:27:40.000Z",
    updatedAt: "2019-07-15T16:27:40.000Z",
  },
  {
    brdGameId: 3,
    name: "Bohnanza",
    minPlayers: "2",
    maxPlayers: "7",
    avgPlayTime: "45",
    description:
      "Bohnanza is the first in the Bohnanza family of games and has been published in several different editions.&#10;&#10;The cards are colorful depictions of beans in various descriptive poses, and the object is to make coins by planting fields (sets) of these beans and then harvesting them.  To help players match their cards up, the game features extensive trading and deal making.  One of the most unique features of the game is that you cannot rearrange your hand, as you need to play the cards in the order that you draw them.&#10;&#10;The original German edition supports 3-5 players.&#10;&#10;The English version from Rio Grande Games comes with the first edition of the first German expansion included in a slightly oversized box.  One difference in the contents, however, is that bean #22's Weinbrandbohne (Brandy Bean) was replaced by the Wachsbohne, or Wax Bean.  This edition includes rules for up to seven players, like the Erweiterungs-Set, but also adapts the two-player rules of Al Cabohne in order to allow two people to play Bohnanza.&#10;&#10;",
    category: "Card Game",
    minAge: "13",
    img:
      "https://cf.geekdo-images.com/original/img/Hwyjq4cYfXY2l07Hm9mncHlrI4U=/0x0/pic69366.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/uwI4ME5bA9KTNqmo6fYdUAV8NkE=/fit-in/200x150/pic69366.jpg",
    bgGeekID: "11",
    createdAt: "2019-07-15T20:32:39.000Z",
    updatedAt: "2019-07-15T21:38:46.550Z",
  },
  {
    brdGameId: 4,
    name: "Twilight Imperium",
    minPlayers: "2",
    maxPlayers: "6",
    avgPlayTime: "240",
    description:
      "Sprawling game of space faring races out to dominate known space.  This game combines a number of elements from other game systems into a unique package.  Hex tiles create the game board and each player begins with a race with unique powers.  The game focuses heavily on a political voting phase and a web of trade agreements between players.&#10;&#10;Part of the Twilight Imperium Series / Family.&#10;&#10;Expanded with:&#10;&#10;    Twilight Imperium: Borderlands&#10;    Twilight Imperium: Distant Suns&#10;    Twilight Imperium: Twilight Armada&#10;    Twilight Imperium: The Outer Rim&#10;&#10;&#10;&#10;&#10;Re-implemented in:&#10;&#10;    Twilight Imperium (Second Edition)&#10;    Twilight Imperium (Third Edition)&#10;    Twilight Imperium (Fourth Edition)&#10;&#10;&#10;&#10;&#10;",
    category: "Civilization",
    minAge: "12",
    img:
      "https://cf.geekdo-images.com/original/img/COM3xWJeoqoM5PXSwjRCjujM1cI=/0x0/pic857.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/5ub28oCmkGHYgt-d00RkEgX5AQU=/fit-in/200x150/pic857.jpg",
    bgGeekID: "24",
    createdAt: "2019-07-15T20:32:39.000Z",
    updatedAt: "2019-07-15T21:39:18.073Z",
  },
  {
    brdGameId: 5,
    name: "RoboRally",
    minPlayers: "2",
    maxPlayers: "8",
    avgPlayTime: "120",
    description:
      "The robots of the Robo Rally automobile factory spend their weekdays toiling at the assembly line. They put in hard hours building high-speed supercars they never get to see in action. But on Saturday nights, the factory becomes a world of mad machines and dangerous schemes as these robots engage in their own epic race.It takes speed, wits, and dirty tricks to become a racing legend! Each player chooses a robot and directs its moves by playing cards. Chaos ensues as all players reveal the cards they've chosen. Players face obstacles like industrial lasers, gaping pits, and moving conveyor belts -- but those can also be used to their advantage! Each player aims to make it to each of the checkpoints in numerical order. The first player to reach all of the checkpoints wins. (source: http://avalonhill.wizards.com/games/robo-rally/comingsoon)&#10;&#10;&#10;&#10;In RoboRally players each control a different robot in a race through a dangerous factory floor. Several goals will be placed on the board and you must navigate your robot to them in a specific order. The boards can be combined in several different ways to accommodate different player counts and races can be as long or as short as player's desire.&#10;&#10;In general, players will first fill all of their robot's &quot;registers&quot; with facedown movement cards. This happens simultaneously and there is a time element involved. If you don't act fast enough you are forced to place cards randomly to fill the rest. Then, starting with the first register, everyone reveals their card. The card with the highest number moves first. After everyone resolves their movement they reveal the next card and so on. Examples of movement cards may be to turn 90 degrees left or right, move forward 2 spaces, or move backward 1 space though there are a bigger variety than that. You can plan a perfect route, but if another robot runs into you it can push you off course. This can be disastrous since you can't reprogram any cards to fix it!&#10;&#10;Robots fire lasers and factory elements resolve after each movement and robots may become damaged. If they take enough damage certain movement cards become fixed and can no longer be changed. If they take more they may be destroyed entirely. The first robot to claim all the goals in the correct order wins, though some may award points and play tournament style.&#10;&#10;The game was reprinted by Avalon Hill (Hasbro/WotC) in 2005.&#10;&#10;",
    category: "Maze",
    minAge: "12",
    img:
      "https://cf.geekdo-images.com/original/img/6Qod68Kgmb6NgXWdOglFnfAZiiE=/0x0/pic1000553.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/4zQbF0XfyuNPaiv8rPLyFV-Sbl8=/fit-in/200x150/pic1000553.jpg",
    bgGeekID: "18",
    createdAt: "2019-07-15T20:32:39.000Z",
    updatedAt: "2019-07-15T21:42:45.495Z",
  },
  {
    brdGameId: 6,
    name: "Dune",
    minPlayers: "2",
    maxPlayers: "6",
    avgPlayTime: "180",
    description:
      "Set thousands of years in the future, Dune the board game is based on the Frank Herbert novels about an arid planet at the heart of the human space empire's political machinations.&#10;&#10;Designed by the creators at Eon of 'Cosmic Encounter fame, some contend that the game can best be described as Cosmic Encounter set within the Dune universe, but the two games bear little in common in the actual mechanisms or goals; they're just both set in space. Like Cosmic Encounter, it is a game that generates player interaction through negotiation and bluffing.&#10;&#10;Players each take the role of one of the factions attempting to control Dune. Each faction has special powers that overlook certain rules in the game. Each turn players move about the map attempting to pick up valuable spice while dealing with giant sandworms, deadly storms, and other players' military forces. A delicate political balance is formed amongst the factions to prevent any one side from becoming too strong. When a challenge is made in a territory, combat takes the form of hidden bids with additional treachery cards to further the uncertainty.&#10;&#10;The game concludes when one faction (or two allied factions) is able to control a certain number of strongholds on the planet.&#10;&#10;Note that the Descartes edition of Dune includes the Duel Expansion and Spice Harvest Expansion, the &quot;Landsraad variant from Avalon Hill's General magazine, and additional character disks not provided by AH.&#10;&#10;",
    category: "Bluffing",
    minAge: "12",
    img:
      "https://cf.geekdo-images.com/original/img/d7LVl0jTBxCGrtiCZnUWFj3LaIc=/0x0/pic279251.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/0P89z7y1-5jZ4jnex9MfdOCTTN8=/fit-in/200x150/pic279251.jpg",
    bgGeekID: "121",
    createdAt: "2019-07-15T20:32:39.000Z",
    updatedAt: "2019-07-16T15:52:48.269Z",
  },
  {
    brdGameId: 7,
    name: "Dune",
    minPlayers: "2",
    maxPlayers: "4",
    avgPlayTime: "90",
    description:
      "This game should not be confused with the grail game Dune which has the same name (but is based on the book rather than the movie).&#10;&#10;Based on the movie, this version of Dune features photos of the stars on pawns divided into teams of three. Each character has its own strength and guile values. Players can move around the outer desert spaces to harvest monetary units of spice or can move around the inner castle spaces to build up strength. Players can use spice to buy random equipment cards, spice harvesters, or extra boosts of guile when under attack. Players can also invest in the craps-like commodity markets that pay off on certain dice rolls. The artwork is slick, the rules are relatively simple compared to AH's version, and games go fairly quickly since all fights are to the death.&#10;&#10;",
    category: "Movies / TV / Radio theme",
    minAge: "10",
    img:
      "https://cf.geekdo-images.com/original/img/KZdLiDVjg3lhgU4Xq5WucCPLn6s=/0x0/pic242509.jpg",
    thumbnail:
      "https://cf.geekdo-images.com/thumb/img/v31rgJ3lS4HhlfCernV2owQRsJA=/fit-in/200x150/pic242509.jpg",
    bgGeekID: "680",
    createdAt: "2019-07-16T16:35:47.000Z",
    updatedAt: "2019-07-16T16:35:49.060Z",
  },
]
