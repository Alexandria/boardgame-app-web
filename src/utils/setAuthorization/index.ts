import axios from "axios"

// Define
type SetAuthorization = (token: string) => void

// Implement
export const setAuthorization: SetAuthorization = token => {
  if (token) {
    axios.defaults.headers.common["Authorization"] = `Bearer ${token}`
  } else {
    delete axios.defaults.headers.common["Authorization"]
  }
}
