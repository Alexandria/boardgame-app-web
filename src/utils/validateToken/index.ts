// Define
import * as jwt from "jsonwebtoken"

type ValidateToken = <A>(a: A) => A

// Implement
export const validateToken: ValidateToken = x => x

//https://www.youtube.com/watch?v=8Ip0pcwbWYM&list=PL55RiY5tL51q4D-B63KBnygU6opNPFk_q&index=13
export const verifyToken = (token: string) => {
  try {
    const key = process.env.REACT_APP_SECRET_KEY
      ? process.env.REACT_APP_SECRET_KEY
      : ""
    return jwt.verify(token, key)
  } catch (err) {
    return err.message
  }
}
