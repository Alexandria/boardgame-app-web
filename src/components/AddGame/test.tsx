import { AddGame } from "."
import React from "react"
import { shallowRender } from "../../utils/shallowRender"

describe("AddGame", () => {
  it("renders", () => {
    // Arrange
    const valA = AddGame

    // Act
    const received = shallowRender(<AddGame>{valA}</AddGame>)

    // Assert
    expect(received).toMatchSnapshot()
  })
})
