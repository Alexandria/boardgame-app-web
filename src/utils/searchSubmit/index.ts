import { FormikActions } from "formik"
import Axios from "axios"
import { validateToken } from "../validateToken"
import xml, { ElementCompact } from "xml-js"

// Define

export const options = {
  compact: true,
  ignoreComment: true,
  alwaysChildren: true,
  ignoreDeclaration: true,
}

export interface Item {
  name: {
    _attributes: {
      value: string
    }
  }
  yearpublished: {
    _attributes: {
      value: string
    }
  }
  _attributes: {
    id: string
  }
}

export const ItemInit = [
  {
    name: {
      _attributes: {
        value: " ",
      },
    },
    yearpublished: {
      _attributes: {
        value: " ",
      },
    },
    _attributes: {
      id: " ",
    },
  },
]
export type SearchSubmit = (
  values: { search: string },
  actions: FormikActions<{ search: string }>,
) => Promise<void | Item[] | undefined>

// Implement
export async function searchSubmit(
  values: {
    search: string
  },
  actions: FormikActions<{
    search: string
  }>,
) {
  const { setStatus, setSubmitting } = actions
  const { search } = values
  setStatus(null)
  //Set up await syntax
  if (validateToken(localStorage.jwtToken)) {
    const result = await Axios.get(
      `http://localhost:8080/https://www.boardgamegeek.com/xmlapi2/search?query=${search}&type=boardgame`,
    )
    const data: ElementCompact = xml.xml2js(result.data, options)

    if (data.items.item) {
      const results: Item[] = data.items.item.length
        ? data.items.item.map((game: Item) => game)
        : data.items.item
      setStatus("Done! Check console for results")
      setSubmitting(false)
      const length = results.length ? results.length : 1
      console.log(
        "Result====> SearchSubmit Utility...Length:",
        length,
        " Result!",
        results,
      )
      return results
    } else {
      setStatus(`No results found containing the name "${search}"`)
      setSubmitting(false)
      return null
    }
  } else {
    setStatus("Please login to add a game to your collection")
    setSubmitting(false)
    // props.history.push("/login")
    return null
  }
}
