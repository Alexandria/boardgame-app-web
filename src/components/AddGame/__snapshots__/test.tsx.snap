// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddGame renders 1`] = `
<div>
  <SearchBar
    onsubmit={[Function]}
  />
  <h2>
    Found: 
    0
     Games
  </h2>
  <ResultList
    items={Array []}
  />
</div>
`;
